src/main/java/org/opentripplanner/middleware/docs/PublicApiDocGenerator.java

```diff
@@ -0,0 +1,416 @@
+    private static final String AWS_API_STAGE = getConfigPropertyAsText("AWS_API_STAGE");
+
+    // Cached variables.
+    private final ObjectNode rootNode;
+    private final JsonNode templateJson;
+    private final ObjectNode pathsNode;
+    private final ObjectNode definitionsNode;
+    private final ArrayNode tagsNode;
+
+    public PublicApiDocGenerator() throws IOException, InterruptedException {
+        // Start with the swagger YAML skeleton generated by spark-swagger (clone it)
+        String autoGeneratedSwagger = getSwaggerDocsAsString();
+
+        // The document we are working off and related nodes.
+        rootNode =  YamlUtils.yamlMapper.readTree(autoGeneratedSwagger).deepCopy();
+        definitionsNode = (ObjectNode) rootNode.get("definitions");
+        pathsNode = (ObjectNode) rootNode.get("paths");
+        tagsNode = (ArrayNode) rootNode.get("tags");
+
+        // Load definitions from a template file.
+        templateJson = YamlUtils.yamlMapper.readTree(OtpMiddlewareMain.class.getClassLoader().getResourceAsStream(API_TEMPLATE_FILE));
+    }
+
+    /**
+     * Generate an enhanced public-facing Swagger document from the one auto-generated by spark-swagger.
+     * This handles things like adding security parameters, method responses, and removing restricted API resources
+     * from the public-facing document.
+     */
+    public Path generatePublicApiDocs() throws IOException {
+
+        // Do the modifications for creating a public API documentation.
+        insertPeliasGeocoderEndpoint(templateJson);
+        removeRestrictedPathsAndTags();
+        addAuthorizationParams(templateJson.get("securityDefinitions"));
+        insertMethodResponses(templateJson.get("responses"));
+        insertAbstractUserRefs(templateJson.get("definitions"));
+        inlineArrayDefinitions();
+        // TODO: Add description to the fields of generated types?
+
+        // Cleanup the final document.
+        generateMissingTypes();
+        removeUnusedTypes();
+        alphabetizeEntries(pathsNode);
+        alphabetizeEntries(definitionsNode);
+
+        // Overwrite top-level parameters.
+        rootNode.put("host", AWS_API_SERVER);
+        rootNode.put("basePath", "/" + AWS_API_STAGE);
+        ((ObjectNode) rootNode.get("info")).put("version", getVersionFromJar());
+
+
+        // Generate output file.
+        Path outputPath = new File(OUTPUT_FILE).toPath();
+        String yamlOutput = YamlUtils.yamlMapper.writer().writeValueAsString(rootNode);
+        Files.writeString(outputPath, yamlOutput);
+
+        LOG.info("Wrote public API documentation to: {}", outputPath);
+
+        return outputPath;
+    }
+
+    /**
+     * Extracts the version number from the JAR file name
+     * (modified from https://stackoverflow.com/questions/14189162/get-name-of-running-jar-or-exe#19045510).
+     */
+    private String getVersionFromJar() {
```

Comment:
This method can be static.

I'm not sure we should rely on the jar file for this. See Data Tools for a method that shows how it extracts git properties https://github.com/ibi-group/datatools-server/blob/9f74b821cf351efcdaf7c9c93a3ae8b694d3c3b1/src/main/java/com/conveyal/datatools/manager/DataManager.java#L181-L212. This can be non-blocking, but it might be worth adding a TODO about this. Also, we may want to share this info elsewhere in the app, so it might be worth moving this to ConfigUtils.

Reply:
Method moved and comment updated in https://github.com/ibi-group/otp-middleware/pull/59/commits/aee52b65a0058fc403f8e00a8c7699babd0398ed

Reply:
Method moved and TODO added in https://github.com/ibi-group/otp-middleware/pull/59/commits/4e3253e4b58640c8a42db4a24d02ac2c3319a3a7